lib/pc_util.c

```diff
@@ -263,3 +288,80 @@ void pc_bounds_merge(PCBOUNDS *b1, const PCBOUNDS *b2)
 	if ( b2->ymax > b1->ymax ) b1->ymax = b2->ymax;
 }
 
+uint8_t *
+pc_bounds_xyzm_to_bounding_diagonal_wkb(
+	const PCSTATS *stats, const PCBOUNDS *bounds, const PCSCHEMA *schema,
+	size_t *wkbsize)
```

Comment:
Agree. How about having two functions, `pc_bounds_to_bounding_diagonal_wkb` and `pc_bounds_xyzm_bounding_diagonal_wkb`, and use the former as an optimization for the 2D case? See the updated patch.